Gopkg.toml

```diff
@@ -0,0 +1,102 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+
+[[constraint]]
+  name = "github.com/blang/semver"
+  version = "3.5.0"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/docker/machine"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/jimmidyson/go-download"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/johanneswuerbach/nfsexports"
+
+[[constraint]]
+  name = "github.com/libvirt/libvirt-go"
+  version = "3.4.0"
+
+[[constraint]]
+  name = "github.com/moby/hyperkit"
+  version = "0.20171020.0"
+
+[[constraint]]
+  name = "github.com/pkg/errors"
+  version = "0.8.0"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/r2d4/external-storage"
+
+[[constraint]]
+  name = "github.com/spf13/pflag"
+  version = "1.0.1"
+
+[[constraint]]
+  name = "github.com/spf13/viper"
+  version = "1.0.0"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/zchee/go-vmnet"
+
+[[constraint]]
+  name = "k8s.io/api"
+  version = "kubernetes-1.11.2"
+
+[[constraint]]
+  name = "k8s.io/apimachinery"
+  version = "kubernetes-1.11.2"
+
+[[constraint]]
+  name = "k8s.io/client-go"
+  version = "7.0.0"
+
+[[constraint]]
+  name = "k8s.io/kubernetes"
+  version = "1.10.0"
+
+[[override]]
+  name = "k8s.io/api"
+  version = "kubernetes-1.10.0"
```

Comment:
@r2d4 so when I do `dep init` it always picks the `1.11.2` but after the vendor when you try to compile it always fails with an error. So that is the reason I need to override it again.

Reply:
I got it all workin with 1.11 with this TOML:
https://gist.github.com/dlorenc/07b36d97d77b5d06299307f43d2a6167

Reply:
@dlorenc why you need to override the same version if it is part of constraint like https://gist.github.com/dlorenc/07b36d97d77b5d06299307f43d2a6167#file-gopkg-toml-L72-L78 and https://gist.github.com/dlorenc/07b36d97d77b5d06299307f43d2a6167#file-gopkg-toml-L88-L94 can you try without override because that what I tried.

Reply:
Not sure what you mean, we need override for non-direct dependencies and constraint for direct ones.

There are some bugs with dep and client-go, so including everything is basically the only way I've ever seen it work.

When you bump to 1.11, you also have to add an override for:

```
[[override]]
  name = "github.com/json-iterator/go"
  version = "1.1.3-22-gf2b4162"
```

Reply:
@dlorenc I don't think you need to use different for direct and non-direct dependency as per doc https://golang.github.io/dep/docs/Gopkg.toml.html#override this is used only when `dependencies, direct and transitive, and supersedes all other [[constraint]] declarations for that project. ` but I am not much expert in that side :/